code/modules/mob/living/carbon/xenomorph/abilities/ravager/ravager_powers.dm

```diff
@@ -333,6 +318,8 @@
 
 	// Heal
 	X.gain_health(heal_amount)
+	X.gain_health((Clamp(heal_per_rage / 100 * (X.maxHealth - X.health), 100, 100)))
+
```

Comment:
> I don't know what you're even trying to do here. So the first line `X.gain_health(heal_amount)` is either going to heal 200 or 0, and then the rest is _always_ going to heal for 100 because you're clamping the value between 100 and 100. So if it just always wanted it to heal for 100 then it should be `X.gain_health(100)` If you wanted it to be a heal that scales based on amount of rage then you'd do something like:
> 
> ```
> if(heal_amount > 0)
> 	heal_amount += heal_per_rage // Because the rage value was already decreased above
> heal_amount += heal_per_rage * BD.rage
> X.gain_health(heal_amount)
> ```
> 
> But this is likely going to heal far more than you want.
> 
> Also, unless I'm really not understanding how it works, gain_health is already going to cap the healing to their brute/fire damage ~(though I don't know why it only heals fire if there's brute damage lol)~.

turns out all i needed to really do was just make it the x.gain_health(100) LOL. More sane now at least, TY